### PR TITLE
raftstore: report server is busy

### PR DESCRIPTION
**Pull request by BusyJay**
ref pingcap/pd#414 pingcap/kvproto#123

@siddontang @huachaohuang PTAL

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: src/raftstore/store/store.rs [hunk 1]
```diff
@@ -700,8 +702,13 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             self.on_ready_result(region_id, res)
         }
 
+        let dur = duration_to_nanos(t.elapsed());
+        if dur >= self.cfg.raft_base_tick_interval * self.cfg.raft_election_timeout_ticks as u64 {
```
Comment (zhangjinpeng1987) [inline src/raftstore/store/store.rs]: ?

**Comment by zhangjinpeng1987** [comment]
LGTM

File: src/raftstore/store/store.rs [hunk 2]
```diff
@@ -701,8 +703,18 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             self.on_ready_result(region_id, res)
         }
 
+        let dur = t.elapsed();
+        if !self.is_busy {
+            let election_timeout =
+                Duration::from_millis(self.cfg.raft_base_tick_interval *
+                                      self.cfg.raft_election_timeout_ticks as u64);
+            if dur >= election_timeout {
```
Comment (huachaohuang) [inline src/raftstore/store/store.rs]: Don't need to clear the flag if `dur < election_timeout`?

Comment (BusyJay) [inline src/raftstore/store/store.rs]: It will be cleared after reporting to pd.

**Comment by huachaohuang** [comment]
LGTM

**Pull request merged**